src/payment-information/payment-information.service.ts

```diff
@@ -0,0 +1,149 @@
+import { Injectable, Logger, NotFoundException } from '@nestjs/common';
+import { CreateCCInformationInput } from './dto/create-creadit-card-information.input';
+import { InjectModel } from '@nestjs/mongoose';
+import { PaymentInformation } from './entities/payment-information.entity';
+import { Model } from 'mongoose';
+import { PaymentMethod } from 'src/payment-method/payment-method.enum';
+import { User } from 'src/graphql-types/user.entity';
+import { UserPaymentInformation } from './entities/user-payment-information.entity';
+
+@Injectable()
+export class PaymentInformationService {
+  constructor(
+    @InjectModel(PaymentInformation.name)
+    private paymentInformationModel: Model<PaymentInformation>,
+    // initialize logger with service context
+    private readonly logger: Logger,
+  ) {}
+
+  /**
+   * Creates a credit card payment information for a user.
+   * @param ccInput - The input for creating the credit card information.
+   * @param user - The user for whom the payment information is being created.
+   * @returns A promise that resolves to the created payment information.
+   */
+  createCreditCardInformation(
+    ccInput: CreateCCInformationInput,
+    user: User,
+  ): Promise<UserPaymentInformation> {
+    this.logger.log(`{createCreditCardInformation} input: ${ccInput}`);
+
+    // build paymentInformation
+    return this.paymentInformationModel.create({
+      paymentMethod: PaymentMethod.CREDIT_CARD,
+      methodDetails: ccInput,
+      user,
+    });
+  }
+
+  /**
+   * Creates a new payment information record that requires no additional information.
+   *
+   * @param paymentMethod - The payment method to be associated with the payment information.
+   * @param user - The user for whom the payment information is being created.
+   * @returns A Promise that resolves to the created PaymentInformation object.
+   */
+  createPaymentInformation(
```

Comment:
This will be called upon user creation events.
Every user will get prepayment and invoice added automatically.
For these, we need no information from the user.